### ISSUE TITLE
Error reading labels in keypoints training

### ISSUE THREAD

**Issue by abelBEDOYA** [ultralytics/ultralytics#4522]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

Hello, 

I'm trying to train a yolo pose model. I have already got my labeled dataset and I've set it properly. The train.yaml looks like this:
![Screenshot from 2023-08-23 16-44-31](https://github.com/ultralytics/ultralytics/assets/89113493/4a909b33-c048-462f-be1b-c793be370db9)

I launch the trainig this way:
`yolo pose train data=/app/datasets/bisagra_keypoints/train.yaml model=yolov8n-pose.pt epochs=100 imgsz=640`

But i got this error:

![Screenshot from 2023-08-23 16-41-56](https://github.com/ultralytics/ultralytics/assets/89113493/ffbc6a80-7d18-42ad-939e-51510b961509)

The train.yaml is found correctly, and the image paths (train and val) as well. I have no clue why I'm running into this error.

### Environment

yolov8 ultralytics 8.0.145 (8.0.160 the error still shows up)
Ubuntu 20.04
Python 3.8.10
torch 1.13.1
numpy 1.24.2
Cuda compilation tools, release 11.6, V11.6.124
Build cuda_11.6.r11.6/compiler.31057947_0

### Minimal Reproducible Example

`yolo pose train data=/app/datasets/bisagra_keypoints/train.yaml model=yolov8n-pose.pt epochs=100 imgsz=640`

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@abelBEDOYA see https://docs.ultralytics.com/datasets/pose/ for example Pose dataset formats, and start by training a default pose dataset and then viewing it's dataset format to replicate the structure:

```
yolo pose train model=yolov8n-pose.pt data=coco8-pose.yaml
```

**Comment by abelBEDOYA** [comment]
@glenn-jocher  thanks you for the quick answer. I've paid attention to the the folder tree and how it is defined in coco8-pose.yaml and it works.

**Solution:** 

train: train/images             ->       train: ./images/train
val: validation/images       ->       val: ./images/validation

On the other hand, I was wondering,  how many images (approximately) are they needed to get good results training keypoints detection?

Thanks again!

**Comment by glenn-jocher** [comment]
@abelBEDOYA I'm glad to hear the solution worked for you.

Regarding your question about the number of images needed for good results in keypoint detection — there isn't a set number, as it can vary greatly depending on the complexity of the task at hand. However, generally speaking, the more training data you have, the better the results tend to be. 

That said, expecting good results in the range of thousands to tens of thousands of images isn't unreasonable. You might find success with fewer, though, if your use case is relatively simple.

And remember, the quality of the data and the annotations are equally, if not more, important than the quantity of the data.

I hope this helps, and best of luck with your project!
